src/middlewares/createInsightsMiddleware.ts

```diff
@@ -76,28 +85,19 @@ aa('setUserToken', 'your-user-token');
           .getHelper()!
           .setQueryParameter('clickAnalytics', true);
 
-        if (hasInsightsClient) {
+        const anonymousUserToken = getInsightsAnonymousUserTokenInternal();
+        if (hasInsightsClient && anonymousUserToken) {
           // When `aa('init', { ... })` is called, it creates an anonymous user token in cookie.
           // We can set it as userToken.
-          setUserTokenToSearch(getInsightsAnonymousUserTokenInternal());
+          setUserTokenToSearch(anonymousUserToken);
+        }
+
+        if (queuedUserToken) {
+          insightsClient('setUserToken', queuedUserToken);
         }
 
-        if (Array.isArray((insightsClient as any).queue)) {
```

Comment:
This part has been move to the line 44. And there it doesn't call `setUserTokenToSearch()`. It instead sets a variable there, and later on, at the line 95, it reads the variable and sets it.